pylanetary/spectral/core.py

```diff
@@ -0,0 +1,808 @@
+    showplots : bool, optional
+          The indicator of whether to plot the best fit line shape or a comparison of line shapes. Default is True, which produces plots.
+    plot_name : string, optional
+          The name of a saved figure. Default is None, which does not save a plot.
+    initial_fit_guess: string, optional.
+          The path to a saved LMFIT result to use as the initial guess. Default is None; the LMFIT .guess method is used. 
+    
+    Returns
+    -------
+    peak : float
+          The frequency or wavelength value of the line peak, which is LMFIT's 'center' parameter.
+    err : float
+          The standard deviation of the peak frequency or wavelength value.
+    chi :
+          The calculated Chi-squared statistic of the line fit.
+    chi_red :
+          The calculated reduced Chi-squared statistic of the line fit.
+    
+    Notes
+    -----
+    Guide to keys in the nested model dictionary:
+          model[] : 
+                The Model name, taken from the user input.
+          ref : 
+                A short reference name of model type.
+          lmfit_func: 
+                The built-in LMFIT function name, needed to properly interface with the LMFIT package.
+          label :
+                The label for the legend included in the final plots.
+          color : 
+                The color of the line for a given fit type when plotting.
+    
+    An initial fit guess is not necessary, but can be very beneficial if the spectra are noisy. If your fit is failing, try including an intial guess to improve the performance of LMFIT. It is recommended that you use a pixel with as small a Doppler shift as possible to make your initial guess save file.
+    
+    '''
+    
+    if fit_type != 'All':
+      if initial_fit_guess == None:
+        pars = model[fit_type]['lmfit_func'].guess(self.spec,x=self.xaxis)
+      else:
+        inital = load_modelresult(initial_fit_guess)
+        pars = inital.params
+      fit_result = model[fit_type]['lmfit_func'].fit(self.spec,pars,x=self.xaxis,scale_covar=False)
+      print(fit_result.fit_report())
+    elif fit_type == 'All':
+      fit_result = {}
+      for key in model:
+        pars = model[key]['lmfit_func'].guess(self.spec,x=self.xaxis)
+        result = model[key]['lmfit_func'].fit(self.spec,pars,x=self.xaxis,scale_covar=False)
+        print(result.fit_report())
+        fit_result[key] = result
+    else:
+      raise ValueError('fit_type must be one of the following options: Gaussian, Lorentzian, Voigt, PseudoV, Moffat, or All')
+    
+    if fit_type != 'All':
+      chi = np.sum(((np.subtract(self.spec,fit_result.best_fit)/self.RMS))**2)
+      chi_red = chi/(self.chan_num - 4)
+    else:
+      chi = np.NAN
+      chi_red = np.NAN
+
+    if showplots is True:
+      test.plot_lines(line_result = fit_result, model_key = fit_type)
+    
+    if plot_name is not None:
+      plt.savefig(plot_name)
```

Comment:
this is going to fail, because there will be no plt object to save.  the plt.savefig must go before show().  Also, there should be a way to save the figure without plotting it, e.g,

fig, ax  = whatever
plot_stuff(ax)
if plot_name is not None:
    plt.savefig(plot_name, dpi=300)
if showplots:
    plt.show()
plt.close()